### PR TITLE
allow user to specify cycles to skip convergence

### PR DESCRIPTION
**Pull request by onufer**
## Description

allow user to specify cycles to skip convergence

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/operators/operator.py [hunk 1]
```diff
@@ -665,7 +674,14 @@ def _checkTightCouplingConvergence(self, activeInterfaces: list):
                 self._convergenceSummary[key].append(coupler.eps)
 
         reportingUtils.writeTightCouplingConvergenceSummary(self._convergenceSummary)
-        return all(converged)
+        converged = all(converged)
+        if converged:
+            runLog.important(
```
Comment (albeanth) [inline armi/operators/operator.py:679]: Just one comment on these `runLog`s. In a draft PR, I have suggested these `runLog`s. I think they'd work well here.  

https://github.com/terrapower/armi/pull/1172/commits/01a08248cd9693c2d890cd81c5a111a660a9edaa

File: armi/operators/tests/test_operators.py [hunk 2]
```diff
@@ -109,11 +109,27 @@ def test_couplingIsActive(self):
...
         with directoryChangers.TemporaryDirectoryChanger():
+            # test cases where we should do a coupling interaction
+            self.o.cs["cyclesSkipTightCouplingInteraction"] = [1]
             self.o.cs["tightCoupling"] = True
+            hasCouplingIntearction = 1
             self.dbWriteForCoupling(writeDB=True)
+            self.assertEqual(self.r.core.p.coupledIteration, hasCouplingIntearction)
+            self.r.core.p.coupledIteration = 0
             self.dbWriteForCoupling(writeDB=False)
+            self.assertEqual(self.r.core.p.coupledIteration, hasCouplingIntearction)
```
Comment (opotowsky) [inline armi/operators/tests/test_operators.py:122]:
```suggestion
            hasCouplingInteraction = 1
            self.dbWriteForCoupling(writeDB=True)
            self.assertEqual(self.r.core.p.coupledIteration, hasCouplingInteraction)
            self.r.core.p.coupledIteration = 0
            self.dbWriteForCoupling(writeDB=False)
            self.assertEqual(self.r.core.p.coupledIteration, hasCouplingInteraction)
```

**Review by opotowsky — APPROVED** [review]
Small typo, but this looks good to me

**Review by mgjarrett — APPROVED** [review]

File: armi/operators/operator.py [hunk 3]
```diff
@@ -387,17 +387,26 @@ def _performTightCoupling(self, cycle: int, timeNode: int, writeDB: bool = True)
...
         """
-        if self.couplingIsActive():
+        if not self.couplingIsActive():
+            # no coupling was requested
+            return
+        skipCycles = (int(val) for val in self.cs["cyclesSkipTightCouplingInteraction"])
+        if cycle in skipCycles:
+            runLog.warning(
+                f"interactAllCoupled disabled this cycle ({self.r.p.cycle}) due to "
+                "`cycleTreatTightConverged` setting."
```
Comment (albeanth) [inline armi/operators/operator.py:397]: Typo here. the setting name in the runLog.warning doesn't match the actual setting name.

Comment (albeanth) [reply armi/operators/operator.py:397]: I'll address this in https://github.com/terrapower/armi/pull/1172. No action needed.

**Pull request merged**